### PR TITLE
`azurerm_site_recovery_replicated_vm` - support for the   `unmanaged_disk`,`target_proximity_placement_group_id`, `target_boot_diag_storage_account_id`, `target_capacity_reservation_group_id`, `target_virtual_machine_scale_set_id`,`multi_vm_group_name`,`target_edge_zone` ,`test_network_id`, `network_interface.test_static_ip`, `network_interface.test_subnet_name` and `network_interface.test_public_ip_address_id`properties

### LINKED ISSUE #8465: azurerm_site_recovery_replicated_vm should support proximity placement groups (ppg)
Although the portal doesn't support it, it is possible to do so with the Az powershell tools or Rest API.

Here is an article from MS on supporting it :https://docs.microsoft.com/en-us/azure/site-recovery/how-to-enable-replication-proximity-placement-groups

The ppg id needs to be added during the creation of the replication protected item (**RecoveryProximityPlacementGroupId**) :

```$TempASRJob = New-AzRecoveryServicesAsrReplicationProtectedItem -AzureToAzure -AzureVmId $VM.Id -Name (New-Guid).Guid -ProtectionContainerMapping $EusToWusPCMapping -AzureToAzureDiskReplicationConfiguration $diskconfigs -RecoveryResourceGroupId $RecoveryRG.ResourceId -RecoveryProximityPlacementGroupId $targetPpg.Id```

Currently trying to replicate any VMs in an AS and PPG causes the following error:
```Code="150373" Message="Enable Replication failed.```

`
{
        "statusMessage": "{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceOperationFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\",\"details\":[{\"code\":\"150373\",\"message\":\"Enable Replication failed. Vm: '/subscriptions/x/resourcegroups/x/providers/microsoft.compute/virtualmachines/x'\"}]}}",
        "eventCategory": "Administrative"
    }
`

### LINKED ISSUE #19846: Support for Test Failover Network and Test failover Settings for ASR
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The following providers are missing Failover Test Network/Failover Test Settings available in the portal. 
azurerm_site_recovery_network_mapping
azurerm_site_recovery_replicated_vm

Here is a screenshot of what im talking about:
Screenshot 2023-01-03 at 12 17 29 PM

### New or Affected Resource(s)/Data Source(s)

azurerm_site_recovery_network_mapping,azurerm_site_recovery_replicated_vm

### Potential Terraform Configuration

_No response_

### References

Referenced to question but no request to add this has been made it seems. 
https://github.com/hashicorp/terraform-provider-azurerm/issues/7039

### LINKED ISSUE #19898: azurerm_site_recovery_replicated_vm does not support multiVmGroupName
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.0

### AzureRM Provider Version

3.38.0

### Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replicated_vm 

### Terraform Configuration Files

```hcl
N/A, the issue is there is no property for multivmgroup - we have been using ARM templates for 2 years due to this missing feature.
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

The ARM template for replicated items has supported setting up a multi vm group name (and now id) for some time:

https://learn.microsoft.com/en-us/azure/templates/microsoft.recoveryservices/2021-12-01/vaults/replicationfabrics/replicationprotectioncontainers/replicationprotecteditems?pivots=deployment-language-bicep 

image

image

### Actual Behaviour

We must use ARM template deployment to support multi vm groups - this makes the resource unusable for scenarios like asr for windows failover clusters that must failover as a group.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by ziyeqf**
fix #19898
fix #8465
fix #19846

new added properties:
`unmanaged_disk`
`target_proximity_placement_group_id`
`target_boot_diag_storage_account_id`
`target_capacity_reservation_group_id`
`target_virtual_machine_scale_set_id`
`multi_vm_group_name`
`target_edge_zone`
`test_network_id`
`network_interface.test_static_ip`
`network_interface.test_subnet_name`
`network_interface.test_public_ip_address_id`

Test
---
image

For the failed ones:
`des`  is failed due to timeout, they are also failing in `main` branch, so I assume it should not block this PR?
`withUnmanagedDisks` also failed due to time out.

`withEdgeZones`, seems the available zone on test scription has gone?

### CODE REVIEW

**Comment by ziyeqf** [comment]
convert to draft to run tests after adding tfo settings

**Comment by ziyeqf** [comment]
close and will reopen when the test running done

**Comment by dsiperek-vendavo** [comment]
Anyway this could get released in this Friday's release?

**Comment by ziyeqf** [comment]
@dsiperek-vendavo Sorry but it could not, there are still some work need to do before make this PR active

**Comment by dsiperek-vendavo** [comment]
When do you expect it to be released? This is an active ticket i also opened up with Azure Support.

**Comment by richshadman** [comment]
@ziyeqf this is fantastic forward motion. I do not think the test failover settings in network will be viable unless we add a root property for **test_network_id** (equivalent to **target_network_id**). Are you able to add that change as well?

**Comment by ziyeqf** [comment]
@richshadman Sure, working on that.

**Comment by ziyeqf** [comment]
@dsiperek-vendavo sorry but I can not give a clear expect right now, as you can see it's a XL PR and there are some prepare work not ready..

File: website/docs/r/site_recovery_replicated_vm.html.markdown [hunk 1]
```diff
@@ -248,9 +248,21 @@ The following arguments are supported:
 
 * `managed_disk` - (Optional) One or more `managed_disk` block as defined below. Changing this forces a new resource to be created.
 
+* `unmanaged_disk` - (Optional) One or more `unmanaged_disk` block.
+
+* `target_proximity_placement_group_id` (Optional) Id of Proximity Placement Group the new VM should belong to when a failover is done.
+
+* `target_boot_diag_storage_account_id` - (Optional) Id of the storage account which the new VM should used for boot diagnostic when a failover is done.
```
Comment (katbyte) [inline website/docs/r/site_recovery_replicated_vm.html.markdown:255]: we should avoid shortening words
```suggestion
* `target_boot_diagnostic_storage_account_id` - (Optional) Id of the storage account which the new VM should used for boot diagnostic when a failover is done.
```

File: website/docs/r/site_recovery_replicated_vm.html.markdown [hunk 2]
```diff
@@ -282,6 +306,12 @@ A `network_interface` block supports the following:
 
 * `recovery_public_ip_address_id` - (Optional) Id of the public IP object to use when a failover is done.
 
+* `test_static_ip` - (Optional) Static IP to assign when a test failover is done.
+
+* `test_subnet_name` - (Optional) Name of the subnet to to use when a test failover is done.
+
+* `test_public_ip_address_id` - (Optional) Id of the public IP object to use when a test failover is done.
```
Comment (katbyte) [inline website/docs/r/site_recovery_replicated_vm.html.markdown:313]: should these be clear this is for failove tests?
```suggestion
* `failover_test_static_ip` - (Optional) Static IP to assign when a test failover is done.

* `failover_test_subnet_name` - (Optional) Name of the subnet to to use when a test failover is done.

* `failover_test_public_ip_address_id` - (Optional) Id of the public IP object to use when a test failover is done.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Review by katbyte — APPROVED** [review]
LGTM ☔

**Pull request merged**